specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,222 @@
+# EIP-7594 -- Data Availability Sampling Core
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Custom types](#custom-types)
+- [Configuration](#configuration)
+  - [Data size](#data-size)
+  - [Custody setting](#custody-setting)
+  - [Helper functions](#helper-functions)
+    - [`get_custody_lines`](#get_custody_lines)
+    - [`compute_extended_data`](#compute_extended_data)
+    - [`compute_extended_matrix`](#compute_extended_matrix)
+    - [`get_data_column_sidecars`](#get_data_column_sidecars)
+- [Custody](#custody)
+  - [Custody requirement](#custody-requirement)
+  - [Public, deterministic selection](#public-deterministic-selection)
+- [Peer discovery](#peer-discovery)
+- [Extended data](#extended-data)
+- [Column gossip](#column-gossip)
+  - [Parameters](#parameters)
+  - [Reconstruction and cross-seeding](#reconstruction-and-cross-seeding)
+- [Peer sampling](#peer-sampling)
+- [Peer scoring](#peer-scoring)
+- [DAS providers](#das-providers)
+- [A note on fork choice](#a-note-on-fork-choice)
+- [FAQs](#faqs)
+  - [Row (blob) custody](#row-blob-custody)
+  - [Subnet stability](#subnet-stability)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Custom types
+
+We define the following Python custom types for type hinting and readability:
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `DataColumn` | `List[Cell, MAX_BLOBS_PER_BLOCK]` | The data of each column in EIP-7594 |
+| `ExtendedMatrix` | `List[Cell, MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS]` | The full data of one-dimensional erasure coding extended blobs (in row major format) |
+| `FlatExtendedMatrix` | `List[BLSFieldElement, MAX_BLOBS_PER_BLOCK * FIELD_ELEMENTS_PER_BLOB * NUMBER_OF_COLUMNS]` | The flattened format of `ExtendedMatrix` |
+
+## Configuration
+
+### Data size
+
+| Name | Value | Description |
+| - | - | - |
+| `NUMBER_OF_COLUMNS` | `uint64((FIELD_ELEMENTS_PER_BLOB * 2) // FIELD_ELEMENTS_PER_CELL)` (= 128) | Number of columns in the extended data matrix. |
+
+### Custody setting
+
+| Name | Value | Description |
+| - | - | - |
+| `SAMPLES_PER_SLOT` | `8` | Number of `DataColumn` random samples a node queries per slot |
```

Comment:
Does it make sense to define a config/preset value for the probability of the data being available/unavailable and derive `SAMPLES_PER_SLOT` from that number? I'm not clear on how we arrived at this number.

Reply:
yes this should be related with probablity, however i would prefer to have this as the config/preset constant, probablity analysis vs sample size could be attached with the specs

Reply:
> Does it make sense to define a config/preset value for the probability of the data being available/unavailable and derive `SAMPLES_PER_SLOT` from that number? I'm not clear on how we arrived at this number.

If >50% of the columns are missing, the probability that `is_data_available` is true is simply at most `0.5^(SAMPLE_PER_SLOT)`

We have `8` because there would be expectedly at most 1/256 of the validators that have `is_data_available` to be true when it should be false.

Reply:
I don't think that `SAMPLES_PER_SLOT` should be as low as 8. Sampling isn't the bandwidth bottleneck (distribution through subnets is), and 8 samples leaves a very-far-from-negligible probability of a full node experiencing a safety violation (chain looks available and confirmed, but it's not), *even without a targeted attack*. 
I would advocate for `SAMPLES_PER_SLOT = 16`.

Reply:
Just brainstorming about resiliency without adding significant security risk, some alternatives came to mind:
- Is there any downside of collecting samples from a fixed number of peers? (e.g. getting 16 samples from 8 peers, instead of potentially up to 16 peers) I think the current proposal is more flexible (in terms of param tweaking) and covers sampling against more peers than this, however I think there's a tradeoff on robustness, i.e. successfully getting all samples from all 16 peers could be harder than getting from 8 peers.
- Is there any security risk if we randomly select 8 connected peers that are assigned different columns, rather than selecting 8 columns then look for the peers if we don't already connect to them?

Reply:
> * Is there any downside of collecting samples from a fixed number of peers? (e.g. getting 16 samples from 8 peers, instead of potentially up to 16 peers) I think the current proposal is more flexible (in terms of param tweaking) and covers sampling against more peers than this, however I think there's a tradeoff on robustness, i.e. successfully getting all samples from all 16 peers could be harder than getting from 8 peers.

It depends what kind of security you are concerned with. From the network's perspective, it makes no difference how you get your samples, it is only important that you do. Everyone could get them from a single source and it wouldn't make any difference. 

From an individual node's perspective, it does matter somewhat, because getting them from multiple nodes (assuming they are actually unrelated) would make you less susceptible to targeted release of data. Without getting into details too much, I don't think this is something to be particularly concerned about, and I don't think there is much of a reason to try hard to get each sample from a different peer.

> Is there any security risk if we randomly select 8 connected peers that are assigned different columns, rather than selecting 8 columns then look for the peers if we don't already connect to them?

Yes, there is. The columns you choose *must* be random. You should not treat the node ids of your peers as random.